MapleServer2/Packets/RegionSkillPacket.cs

```diff
@@ -33,16 +33,35 @@ public static PacketWriter Send(int sourceObjectId, CoordS effectCoord, SkillCas
 
         for (int i = 0; i < tileCount; i++)
         {
-            CoordF currentSkillCoord = skillMoves != null ? skillMoves[i].FireOffsetPosition : CoordF.From(0, 0, 0);
-            CoordF castPosition = Block.ClosestBlock(currentSkillCoord + effectCoord.ToFloat());
+            MagicPathMove magicPathMove = skillMoves?[i];
+            if (magicPathMove is null)
+            {
+                continue;
+            }
+
+            CoordF offSetCoord = magicPathMove.FireOffsetPosition;
+
+            // If false, rotate the offset based on the look direction.
+            if (!magicPathMove.IgnoreAdjust)
+            {
+                // Rotate the offset coord based on the look direction
+                CoordF rotatedOffset = CoordF.From(offSetCoord.Length(), lookDirection);
 
-            pWriter.Write(castPosition);
+                // Add the effect coord to the rotated coord
+                offSetCoord = rotatedOffset + effectCoord.ToFloat();
+            }
+            else
+            {
+                offSetCoord += Block.ClosestBlock(effectCoord.ToFloat());
+            }
+
+            pWriter.Write(offSetCoord);
```

Comment:
I'm moving this logic to outside the packet file tomorrow.

Reply:
![image](https://user-images.githubusercontent.com/1348762/151705742-a6967efe-cce0-4f7a-bbb3-4b7c6e461820.png)
don't forget 😆

Reply:
Updated